Document EF request and response members (#880)
edgio-docs/edgio-docs: src/guides/v7/edge-functions.md

#### Supported Methods and Properties {/* request-supported-methods-and-properties */}

- **Headers**: Access the request headers using `request.headers`.
- **Body**: The request body can be read as:
  - **ArrayBuffer**: `await request.arrayBuffer()`
  - **JSON**: `await request.json()`
  - **Text**: `await request.text()`
- **Method**: `request.method` to get the HTTP method of the request.
- **URL**: `request.url` provides the full URL, and `request.path` gives the request path.
- **Cloning**: To clone a request without its body, use `request.cloneWithoutBody()`.

#### Unsupported Methods and Properties {/* request-unsupported-methods-and-properties */}

The following properties and methods from the standard [`Request`](https://developer.mozilla.org/en-US/docs/Web/API/Request) API are not supported:

- `request.blob()`
- `request.cache`
- `request.credentials`
- `request.clone()`
- `request.destination`
- `request.formData()`
- `request.integrity`
- `request.mode`
- `request.redirect`
- `request.referrer`
- `request.referrerPolicy`
- `request.signal`

> **Note**: The above-mentioned unsupported methods and properties will throw an error if attempted to be used.

Reviewer: ```suggestion
<Callout type="info">

  Using an unsupported method or property will throw an error.

</Callout>
```

### Response Class {/* response-class */}

<Callout type="info">

Edge functions use a modified version of the standard [`Response`](https://developer.mozilla.org/en-US/docs/Web/API/Response) API. See the [Unsupported Methods and Properties](#response-unsupported-methods-and-properties) section for more information.

</Callout>

> Origins fetch requests and edge functions return a `Response` instance representing the response. This object provides methods and properties for accessing and setting the response's headers, body, status, and more. A response can be created using the `Response` class or by calling the `fetch()` function. See the [Edge Function Namespace](#edge-function-namespace) section for more information.

Reviewer: ```suggestion
Origin fetch requests and edge functions return a `Response` instance representing the response. This object provides methods and properties for accessing and setting the response's headers, body, status code, and more. Create a response through the `Response` class or by calling the `fetch()` function. See the [Edge Function Namespace](#edge-function-namespace) section for more information.
```

#### Supported Methods and Properties {/* response-supported-methods-and-properties */}

- **Headers**: Access or modify the response headers using `response.headers`.
- **Body**: The response body can be interacted with using:
  - **ArrayBuffer**: `await response.arrayBuffer()`
  - **JSON**: `await response.json()`
  - **Text**: `await response.text()`
- **Status**: `response.status` to get the HTTP status code of the response. `response.statusText` provides the corresponding status text.
- **URL**: `response.url` provides the URL of the response.
- **Redirected**: `response.redirected` is a property that indicates whether the response is a result of a redirection.

>   - **Note**: a response can be redirected up to 5 times before an exception is thrown.

Reviewer: ```suggestion
    <Callout type="info">

      You may redirect a response up to 5 times before an exception is thrown.

    </Callout>
```

  };
  ```

- `process.env` Namespace

  This namespace is not globally available in edge functions. You can use the following polyfill to set environment variables from the context object.

  ```js filename="./polyfills/process-env.js"
  /**
   * Polyfill for process.env.
   */
  global.process = global.process || {env: {}};

  /**
   * Sets environment variables from a given context.
   *
   * @param {Object} context - The context object containing environment variables.
   * @param {Object} context.environmentVars - Key-value pairs of environment variables.
   */
  export function setEnvFromContext({environmentVars}) {
    Object.assign(process.env, environmentVars);
  }
  ```

### Helper Functions {/* helper-functions */}

- `createFetchForOrigin` for the `fetch()` API <a id="createFetchForOrigin"></a>

  This function returns a modified `fetch()` function that includes the origin server. This is useful for making multiple requests to the same origin or overriding the global function.

>   **Note**: Some third-party libraries let you specify a `fetch()` function. If you are unable to set this in your library, you can override the global one using this helper. See the [Origin Requests Using fetch()](#origin-requests-using-fetch) section for more details.

Reviewer: ```suggestion
    <Callout type="info">

      Some third-party libraries let you specify a `fetch()` function. If you are unable to set this in your library, you can override the global one using this helper. See the [Origin Requests Using fetch()](#origin-requests-using-fetch) section for more details.

    </Callout>
```
